### ISSUE TITLE
__namedParameters / param comments doesn't work with `excludeNotExported`

### ISSUE THREAD

**Issue by fc** [TypeStrong/typedoc#1000]
## Expected Behavior
Expect named parameters to be able to have comment docs

## Actual Behavior
Not seeing comments

## Steps to reproduce the bug

Create a class that contains this method:

```
    /**
     * This is a simple member function.
     *
     * It should be inherited by all subclasses.
     *
     * @param {object} __namedParameters Options
     * @param example The new name2
     */
    public setName2({example}: {example: string}) { }

```

But - I can't figure out what in my class is causing the bug since it works in the example.

Funnily enough it *works* when I do this modifying the example in `typedoc/examples/basic/src/classes.ts` but `SomeFile.tsx` it doesn't work.

Question:

Could the `.tsx` be a problem?

Here is the output when using `--json`.

I copied the method that works in the example to my class.

Here is the wrong output generated by my class using excludeNotExported:

```
{
	"id": 57,
	"name": "setName2",
	"kind": 4096,
	"kindString": "Call signature",
	"flags": {},
	"comment": {
		"shortText": "This is a simple member function.",
		"text": "It should be inherited by all subclasses.\n"
	},
	"parameters": [
		{
			"id": 58,
			"name": "__namedParameters",
			"kind": 32768,
			"kindString": "Parameter",
			"flags": {},
			"originalName": "__0",
			"type": {
				"type": "reflection",
				"declaration": {
					"id": 59,
					"name": "__type",
					"kind": 65536,
					"kindString": "Type literal",
					"flags": {},
					"sources": [
						{
							"fileName": "classes.ts",
							"line": 141,
							"character": 20
						}
					]
				}
			}
		}
	],
	"type": {
		"type": "intrinsic",
		"name": "void"
	}
}


```

Here is the correct output generated from the example
```
{
        "id": 57,
        "name": "setName2",
        "kind": 4096,
        "kindString": "Call signature",
        "flags": {},
        "comment": {
                "shortText": "This is a simple member function.",
                "text": "It should be inherited by all subclasses.\n"
        },
        "parameters": [
                {
                        "id": 58,
                        "name": "__namedParameters",
                        "kind": 32768,
                        "kindString": "Parameter",
                        "flags": {},
                        "originalName": "__0",
                        "type": {
                                "type": "reflection",
                                "declaration": {
                                        "id": 59,
                                        "name": "__type",
                                        "kind": 65536,
                                        "kindString": "Type literal",
                                        "flags": {},
                                        "children": [
                                                {
                                                        "id": 60,
                                                        "name": "example",
                                                        "kind": 32,
                                                        "kindString": "Variable",
                                                        "flags": {},
                                                        "sources": [
                                                                {
                                                                        "fileName": "classes.ts",
                                                                        "line": 141,
                                                                        "character": 28
                                                                }
                                                        ],
                                                        "type": {
                                                                "type": "intrinsic",
                                                                "name": "string"
                                                        }
                                                }
                                        ],
                                        "groups": [
                                                {
                                                        "title": "Variables",
                                                        "kind": 32,
                                                        "children": [
                                                                60
                                                        ]
                                                }
                                        ],
                                        "sources": [
                                                {
                                                        "fileName": "classes.ts",
                                                        "line": 141,
                                                        "character": 20
                                                }
                                        ]
                                }
                        }
                }
        ],
        "type": {
                "type": "intrinsic",
                "name": "void"
        }
}



```
## Environment
  - Typedoc version: 0.14.2 (TypeScript 3.2.4)
  - Node.js version: 10.4.1
  - OS: OSX

**Comment by fc** [comment]
Aha!
I narrowed it down. The issue is because I have `excludeNotExported` enabled.

**Comment by fc** [comment]
Repro and more context:
```
git clone https://github.com/fc/typedoc typecode-exclude
cd typecode-exclude
yarn
cd examples/basic
./run
```

These are the 2 changes I made:
https://github.com/fc/typedoc/commit/19d0ae47c317071b39a7957a956640d585fb7edc
https://github.com/fc/typedoc/commit/cae1cb29c416905bdecf0a41755130941396f546

Here is what gets generated (the wrong stuff...):
![image](https://user-images.githubusercontent.com/20962/55201945-86c27500-5182-11e9-880b-cdfec76019c4.png)

Here is what is supposed to be generated (the right stuff...):
![image](https://user-images.githubusercontent.com/20962/55202112-46172b80-5183-11e9-9c90-7aee308d671e.png)

According to the generated json posted in the start of this thread it probably has something to do with the `isExported` flag being set.

**Comment by Gerrit0** [comment]
The exported behavior has been fixed in 0.16. The given example (once `@param` is corrected) now renders as:

![image](https://user-images.githubusercontent.com/19329837/72479053-8dbd8100-37b0-11ea-90d9-10792a28cdf7.png)

**Comment by phihag** [comment]
I believe I'm seeing this bug again in 0.17.4, with the following JavaScript:
```
/**
 * FOO
 * @param {Object} __namedParameters various options
 * @param {Number} foo An interesting value
 * @param {Number} __namedParameters.bar Another value
 */
function demo({ foo=42, bar=43 } = {}) { }
module.exports = {demo};
```
[Here is a repository demonstrating the problem](https://github.com/phihag/typedoc-named-demo).
